assembly/src/parsers/ast/mod.rs

```diff
@@ -46,6 +115,32 @@ pub struct ProcedureAst {
     pub index: u32,
 }
 
+impl ProcedureAst {
+    pub fn write_into(&self, target: &mut ByteWriter) {
+        target.write_string(&self.name);
+        target.write_u32(self.num_locals);
+        target.write_vec_node(&self.body);
+        target.write_bool(self.is_export);
+        target.write_u32(self.index);
```

Comment:
I would write `self.body` as the last thing. In fact, I'd change the order to be:
* index (should probably be a u16)
* name
* is_export
* num_locals (should probably be a u16)
* body